src/Module.purs

```diff
@@ -42,13 +41,14 @@ import Util.Set ((∪))
 parse :: forall a m. MonadError Error m => String -> SParser a -> m a
 parse src = liftEither <<< lmap (E.error <<< show) <<< runParser src
 
-parseProgram :: forall m. LoadFile m => Array Folder -> File -> AffError m (Raw S.Expr × List ModuleName)
-parseProgram folders (File file) =
-   loadFile folders (File (file <> fluidExtension)) >>= flip parse P.program
+parseProgram :: forall m. String -> AffError m (Raw S.Expr × List ModuleName)
+parseProgram fluidSrc = flip parse P.program fluidSrc
 
 datasetAs :: forall m. MonadAff m => MonadError Error m => LoadFile m => Array Folder -> Bind File -> Raw ProgCxt -> m (Raw ProgCxt)
 datasetAs folders (x ↦ file) (ProgCxt r@{ datasets }) = do
-   eα <- (fst <$> parseProgram folders file) >>= desug
+   src <- loadFile folders file
+   s × _ <- parseProgram src
```

Comment:
You could keep these two lines in the slightly more idiomatic form that they were previously.